# Data exploration: CALORIC SUPPLY

* This notebook will explore the data regarding the caloric supply per country.


* It present the following queries:  
  * Average caloric supply increase between 1961-2013  
  * Minimum caloric supply in 1961 and 2013  
  * 10 Countries with lowest caloric supply 1961  
  * 10 Countries with lowest caloric supply 2013
  * 10 Countries with highest caloric supply 2013


#### Connection to DB

In [1]:
import mysql.connector
%load_ext sql
%sql mysql+mysqldb://root:admin@localhost/food_stat

#### Average caloric Supply increase between 1961-2013

In [3]:
%%sql
SELECT
  (SELECT avg(daily_caloric_supply)
   FROM food_sup_life_expectancy
   WHERE YEAR = 2013
     AND daily_caloric_supply IS NOT NULL
     AND code IS NOT NULL ) - avg(daily_caloric_supply) AS 'avg_daily_caloric_increase_2013-1961'
FROM food_sup_life_expectancy
WHERE YEAR = 1961
  AND daily_caloric_supply IS NOT NULL
  AND code IS NOT NULL;

 * mysql+mysqldb://root:***@localhost/food_stat
1 rows affected.


[(Decimal('500.8132'),)]

#### Does the minimum caloric supply between 1961 and 2013 changed?

In [4]:
%%sql
  (SELECT country,
          year,
          daily_caloric_supply
   FROM food_sup_life_expectancy
   WHERE YEAR =1961
     AND daily_caloric_supply IS NOT NULL
   ORDER BY daily_caloric_supply ASC
   LIMIT 1)
UNION
  (SELECT country,
          YEAR,
          daily_caloric_supply
   FROM food_sup_life_expectancy
   WHERE YEAR =2013
     AND daily_caloric_supply IS NOT NULL
   ORDER BY daily_caloric_supply ASC
   LIMIT 1)

 * mysql+mysqldb://root:***@localhost/food_stat
2 rows affected.


[('Burkina Faso', 1961, 1326), ('Central African Republic', 2013, 1879)]

#### 10 countries with lowest daily caloric supply 1961

In [5]:
%%sql
SELECT country, daily_caloric_supply
FROM food_sup_life_expectancy
WHERE year = 1961 and daily_caloric_supply is NOT NULL
ORDER BY daily_caloric_supply ASC
LIMIT 10;

 * mysql+mysqldb://root:***@localhost/food_stat
10 rows affected.


[('Burkina Faso', 1326),
 ('China', 1439),
 ('Myanmar', 1449),
 ('Maldives', 1558),
 ('El Salvador', 1575),
 ('Sudan (former)', 1590),
 ('Niger', 1610),
 ('Djibouti', 1613),
 ('Guinea-Bissau', 1615),
 ('Algeria', 1619)]

#### 10 countries with lowest daily caloric supply 2013

In [8]:
%%sql
SELECT country, daily_caloric_supply
FROM food_sup_life_expectancy
WHERE year = 2013 and daily_caloric_supply is NOT NULL
ORDER BY daily_caloric_supply ASC
LIMIT 10;

 * mysql+mysqldb://root:***@localhost/food_stat
10 rows affected.


[('Central African Republic', 1879),
 ('Zambia', 1930),
 ('Madagascar', 2052),
 ('Afghanistan', 2090),
 ('Haiti', 2091),
 ('North Korea', 2094),
 ('Chad', 2110),
 ('Zimbabwe', 2110),
 ('Uganda', 2130),
 ('Timor', 2131)]

#### 10 countries with the highest daily caloric supply in 2013

In [7]:
%%sql
SELECT country, daily_caloric_supply
FROM food_sup_life_expectancy
WHERE year = 2013 and daily_caloric_supply is NOT NULL
ORDER BY daily_caloric_supply DESC
LIMIT 10;

 * mysql+mysqldb://root:***@localhost/food_stat
10 rows affected.


[('Austria', 3768),
 ('Belgium', 3733),
 ('Turkey', 3706),
 ('United States', 3682),
 ('Northern America', 3663),
 ('Israel', 3610),
 ('Ireland', 3600),
 ('Italy', 3579),
 ('Luxembourg', 3539),
 ('Egypt', 3522)]

#### Conclusion

- The average increase between 1961/2013 was 500 calories/day


- There's an increase in the minimum caloric supply from 1326 calories/day in 1961 to 1879 calories/day in 2013. The increase is a little more than the average in the same period, representing 553 calories/day.


- In 1961 the countries with the lowest daily caloric supply were Burkina Faso, China and Myanmar. Instead in 2013 the countries with the lowest daily caloric supply are the Central African Republic, Zambia and Madagascar. None of the 10 countries with the lowest daily caloric consumption in 1961 are in the same statistics of 2013.